legacy/src/main/java/org/opensearch/sql/legacy/esdomain/LocalClusterState.java

```diff
@@ -132,39 +117,30 @@ public <T> T getSettingValue(Settings.Key key) {
     return (T) latestSettings.getOrDefault(key.getKeyValue(), pluginSettings.getSettingValue(key));
   }
 
-  /** Get field mappings by index expressions. All types and fields are included in response. */
-  public IndexMappings getFieldMappings(String[] indices) {
-    return getFieldMappings(indices, ALL_FIELDS);
-  }
-
   /**
-   * Get field mappings by index expressions, type and field filter. Because
+   * Get field mappings by index expressions. Because
    * IndexMetaData/MappingMetaData is hard to convert to FieldMappingMetaData, custom mapping domain
    * objects are being used here. In future, it should be moved to domain model layer for all
    * OpenSearch specific knowledge.
    *
-   * <p>Note that cluster state may be change inside OpenSearch so it's possible to read different
-   * state in 2 accesses to ClusterService.state() here.
-   *
    * @param indices index name expression
-   * @param fieldFilter field filter predicate
    * @return index mapping(s)
    */
-  private IndexMappings getFieldMappings(
-      String[] indices, Function<String, Predicate<String>> fieldFilter) {
-    Objects.requireNonNull(clusterService, "Cluster service is null");
-    Objects.requireNonNull(resolver, "Index name expression resolver is null");
+  public IndexMappings getFieldMappings(String[] indices) {
+    Objects.requireNonNull(client, "Client is null");
 
     try {
-      ClusterState state = clusterService.state();
-      String[] concreteIndices = resolveIndexExpression(state, indices);
 
-      IndexMappings mappings;
-      if (fieldFilter == ALL_FIELDS) {
-        mappings = findMappingsInCache(state, concreteIndices);
-      } else {
-        mappings = findMappings(state, concreteIndices, fieldFilter);
-      }
+      Map<String, MappingMetadata> mappingMetadata = client.admin()
+              .indices()
+              .prepareGetMappings(indices)
+              .setLocal(true)
```

Comment:
If the data is stale, it will behave as it does today as even today's access is resolved locally.

That being said, it will fail requests with a 4xx as the fields are missing

Reply:
We can accept this as it is the same as V2. Any thoughts on the long-term implications if schema freshness is critical?